seisgo/noise.py

```diff
@@ -39,238 +40,229 @@ def compute_fft(trace,win_len,step,stainv=None,
+                
+                # Check if time difference between two channels are too big. Big start and end time differences will be discarded and move to the next station
+                if abs(sgap)//dt>pad_thre or abs(egap)//dt>pad_thre:
+                    print('Time error for station {0} in {1} is larger than zero padding threshold, check data'.format(ista,sfile))
+                    return None
+                
+                # First zero pad start time side (left)
+                if abs(sgap)%dt>dt/2:
+                    zero=np.zeros(int(abs(sgap)//dt+1))
+                    if sgap<0:        # if tr1 starts earlier, add zero to tr2 on left
+                        tr2.data=np.insert(tr2.data,0,zero)
+                        tr2.stats.starttime=tr1.stats.starttime
+                    else:             # if tr1 starts later, add zero to tr1 on left
+                        tr1.data=np.insert(tr1.data,0,zero)
+                        tr1.stats.starttime=tr2.stats.starttime
+                
+                else:
+                    zero=np.zeros(int(abs(sgap)//dt))
+                    if sgap<0:
+                        tr2.data=np.insert(tr2.data,0,zero)
+                        tr2.stats.starttime=tr1.stats.starttime
+                    else:
+                        tr1.data=np.insert(tr1.data,0,zero)
+                        tr1.stats.starttime=tr2.stats.starttime
+                
+                # Then zero pad end time side (right)
+                if abs(egap)%dt>dt/2:
+                    zero=np.zeros(int(abs(egap)//dt+1))
+                    if egap<0:        # if tr1 ends earlier, add zero to tr1 on right
+                        tr1.data=np.append(tr1.data,zero)
+                    else:             # if tr1 ends later, add zero to tr2 on right
+                        tr2.data=np.append(tr2.data,zero)
+                
+                else:
+                    zero=np.zeros(int(abs(egap)//dt))
+                    if egap<0:
+                        tr1.data=np.append(tr1.data,zero)
+                    else:
+                        tr2.data=np.append(tr2.data,zero)
+                
+                tr1_len=tr1.data.shape
+                tr2_len=tr2.data.shape
+                # if traces have the same length, do orientation correction
+                if  tr1_len==tr2_len:
+                    trN,trE=utils.correct_orientations(tr1,tr2,orient)
+                    print('Orientation correction for {} is finished'.format(ista))
+                else:
+                    print('STOP! Error in pading traces. Check start and end times')
+                    print(tr1_start)
+                    print(tr2_start)
+                    print(tr1_end)
+                    print(tr2_end)
+                    print(tr1.data.shape)
+                    print(tr2.data.shape)
+                    print(sfile,ista)
+                    trE,trN=utils.correct_orientations(tr1,tr2,orient)
```

Comment:
should line 188 be removed? if the lengths are different, I don't think you are supposed to correct the orientation anymore.

Reply:
It is only for debugging. When processing real data, the whole else statement is not expected to execute.

Reply:
but if tr1 and tr2 have different lengths, do you get errors running `trE,trN=utils.correct_orientations(tr1,tr2,orient)`?

Reply:
Yes, that's why I said it's for debugging only.
tr1 and tr2 should have the same length. If they don't something must wrong. Then the error message from correction_orientations() is useful.

Reply:
I can simply remove the else statement since error seldom happens